src/tui.rs

```diff
@@ -1811,9 +1776,134 @@ impl<'a> Context<'a, '_> {
         // current_node will refer to the table. Otherwise, it'll refer to the current row.
         if !matches!(current_node.content, NodeContent::Table(_)) {
             self.block_end();
+            self.table_end_row();
         }
 
         self.block_end(); // table
+        self.table_end_move_focus();
+    }
+
+    fn table_row_move_focus(&mut self) {
+        // Filter down to table rows that are focused.
+        if !self.contains_focus() {
+            return;
+        }
+
+        // Filter down to left/right inputs.
+        if self.input_consumed {
+            return;
+        }
+        let Some(input) = self.input_keyboard else {
+            return;
+        };
+        if !matches!(input, vk::LEFT | vk::RIGHT) {
+            return;
+        }
+
+        let row = self.tree.last_node;
+        let Some(&focused_cell_id) = self.tui.focused_node_path.get(row.borrow().depth + 1) else {
+            return;
+        };
+
+        let mut prev_next = NodeSiblings { prev: None, next: None };
+        let mut focused = None;
+
+        // Iterate through the cells in the row, looking for the focused cell.
+        // Take note of the previous and next focusable cells around the focused one.
+        for cell in Tree::iterate_siblings(row.borrow().children.first) {
+            let n = cell.borrow();
+            if n.id == focused_cell_id {
+                focused = Some(cell);
+            } else if n.attributes.focusable {
+                if focused.is_none() {
+                    prev_next.prev = Some(cell);
+                } else {
+                    prev_next.next = Some(cell);
+                    break;
+                }
+            }
+        }
+
+        if focused.is_none() {
+            return;
+        }
+
+        let forward = input == vk::RIGHT;
+        let children_idx = if forward { NodeChildren::FIRST } else { NodeChildren::LAST };
+        let siblings_idx = if forward { NodeSiblings::NEXT } else { NodeSiblings::PREV };
+        let Some(focused_next) =
+            prev_next.get(siblings_idx).or_else(|| row.borrow().children.get(children_idx))
+        else {
+            return;
+        };
+
+        Tui::build_node_path(Some(focused_next), &mut self.tui.focused_node_path);
+        self.set_input_consumed();
+        self.needs_rerender();
+    }
+
+    fn table_end_move_focus(&mut self) {
```

Comment:
These two functions seem *so similar* that I can't immediately tell the difference except Left/Right vs Up/Down

Reply:
Done. It feels quite nice now IMO.